src/formReducer.js

```diff
@@ -35,6 +38,23 @@ export default function formReducer(state = {}, action) {
       newForms[action.formId] = {}
       return { ...newForms }
 
+    case 'ON_FOCUS':
+      if (!newForms[action.formId]) {
+        newForms[action.formId] = {}
+      }
+      if (!newForms[action.formId].touchedFields) {
+        newForms[action.formId].touchedFields = []
+      }
+
+      newForms[action.formId].touchedFields.push(action.fieldName)
+      const touchedArr = _.uniq(newForms[action.formId].touchedFields)
```

Comment:
Instead of uniq-ing or filtering here, let's do an `if (newForms[action.formId].touchedFields.indexOf(action.fieldName) === -1)`